Assets/Scripts/SelectionManager.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SelectionManager : MonoBehaviour
+{
+
+    [SerializeField] private string selectableTag = "Selectable";
+    [SerializeField] private Material highlightMaterial;
+    [SerializeField] private Material defaultMaterial;
+
+    private Transform _selection;
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
```

Comment:
Instead of changing the material back and forth every update loop u can start by raycasting. Then if the result is null unhighlight your previous saved object. if the result is not null => if the result of the raycast is the same object => do nothing. And lastly if the raycast result is different => unhighlight the previous object, highlight the new one and save the new one in the _selection variable.

Reply:
Im going to be honest with you
I have no idea what you said right there.
Ill try to figure it out